### ISSUE TITLE
QNAME minimization, replacing dnsmasq with Unbound

### ISSUE THREAD

**Issue by cpu** [jlund/streisand-discussions#12]
Suggested by @L-P in https://github.com/jlund/streisand/issues/459

> Hi,
>
> QNAME minimization ([RFC 7816][1]) is a way to resolve domain names without having to send the full request to every authority in the chain.  ie. if you have three authorities, A knows who has `.com`, B `.foo.com`, and C `bar.foo.com`, you ask A for `.com`, B for `foo.com` and C for `bar.foo.com`, instead of leaking the full `bar.foo.com` to everyone.
>
> [1]: https://tools.ietf.org/html/rfc7816
>
> I think this would be a privacy improvement and is quite cheap to implement.
>
> dnsmasq is currently used as a DNS resolver, I propose to replace it with [Unbound](https://unbound.net/) as it implements QNAME minimization.
>
> From the package description:
>
>> Unbound is a recursive-only caching DNS server which can perform DNSSEC
>> validation of results. It implements only a minimal amount of authoritative
>> service to prevent leakage to the root nameservers
>
> FreeBSD considered it stable and secure enough to replace BIND in 2014.
>
> Replacing dnsmasq with Unbound should be painless, migrating a live server could be doable by downloading the unbound package before removing dnsmasq.  I don't know if such a migration should be included in Streisand as it will be dead code as soon as gets written.
> 
> TL;DR questions:
>  1. QNAME minimization, yay or nay?
>  2. Is it worth replacing Dnsmasq ?
>  3. Should an automatic dnsmasq->Unbound migration be included in Streisand?
